doc/page-cmakefunctions.h

```diff
@@ -0,0 +1,131 @@
+
+<hr><br>
+
+\code
+facelift_add_interface(TargetName
+                       INTERFACE_DEFINITION_FOLDER dir
+                       [LINK_LIBRARIES lib1 [lib2...]]
+                       [IMPORT_FOLDERS folder1 [folder2...]])
+\endcode
+
+This function creates a new facelift interface library and generates the corresponding modules (like ipc, types, devtools, etc.)
+
+\param TargetName The name of the interface library that will be created
+\param INTERFACE_DEFINITION_FOLDER The folder with qface definition files
+\param LINK_LIBRARIES List of libraries that will be linked with this interface library
+\param IMPORT_FOLDERS List of folders with qface files that will be imported
+
+<hr><br>
+
+\code
+facelift_generateQRC(OutputFile
+                     INPUT_FOLDERS folder1 [folder2...])
+\endcode
+
+Generates a qrc resource file from the given input folders. This function will look for js, qml, ttf, png, sci, json, astcz, tcsh, tesh, gsh, fsh files.
+
+\param OutputFile The qrc resource file that will be generated
+\param INPUT_FOLDERS List of folders containing resources to be added
+
+<hr><br>
+
+\code
+facelift_add_library(TargetName
+                     [STATIC | SHARED]
+                     [SOURCES src1 [src2...]]
+                     [HEADERS hdr1 [hdr2...]]
+		     [SOURCES_GLOB srcs [*.cpp]]
+		     [HEADERS_GLOB hdrs [*.h]]
+                     [SOURCES_GLOB_RECURSE srcs [*.cpp]]
+                     [HEADERS_GLOB_RECURSE hdrs [*.h]]
+ 		     [HEADERS_NO_INSTALL hdr1 [hdr2...]]
+		     [HEADERS_GLOB_NO_INSTALL hdrs [*.h]]
+		     [HEADERS_GLOB_RECURSE_NO_INSTALL hdrs [*.h]]
+		     [PUBLIC_HEADER_BASE_PATH path]
+                     [LINK_LIBRARIES lib1 [lib2...]]
+                     [NO_EXPORT]
+                     [NO_INSTALL])
+\endcode
+
+This function creates a library  with the name \e TargetName from the specified source and header files.
+
+<hr><br>
+
+\code
+facelift_add_executable(TargetName
+                        [SOURCES src1 [src2...]]
+                        [HEADERS hdr1 [hdr2...]]
+                        [SOURCES_GLOB_RECURSE srcs [*.cpp]]
+	                [HEADERS_GLOB_RECURSE hdrs [*.h]]
+                        [LINK_LIBRARIES lib1 [lib2...]])
+
+\endcode
+
+This function creates an executable with the name \e TargetName from the specified source and header files.
+
+<hr><br>
+
+\code
+facelift_add_test(TestName
+                  [SOURCES src1 [src2...]]
+                  [HEADERS hdr1 [hdr2...]]
+		  [SOURCES_GLOB_RECURSE srcs [*.cpp]]
+                  [HEADERS_GLOB_RECURSE hdrs [*.h]]                  
+                  [LINK_LIBRARIES lib1 [lib2...]])
+\endcode
+
+This function creates a test with the name \e TestName.
+
+<hr>
+
+
+\param TargetName The name of the component that will be created
+\param STATIC If provided, a static library will be created
+\param SHARED If provided, a shared library will be created (default)
```

Comment:
it is not the default. The actual type follows https://cmake.org/cmake/help/v3.0/variable/BUILD_SHARED_LIBS.html